cranelift/codegen/src/isa/aarch64/lower.rs

```diff
@@ -768,14 +768,15 @@ impl LowerBackend for AArch64Backend {
             assert!(op1 == Opcode::Jump);
         }
 
-        if let Ok(()) = super::lower::isle::lower_branch(
+        if let Some(temp_regs) = super::lower::isle::lower_branch(
             ctx,
             &self.triple,
             &self.flags,
             &self.isa_flags,
             branches[0],
             targets,
         ) {
+            assert!(temp_regs.len() == 0);
```

Comment:
Is there any case where lower_branch is expected to return temporary registers? If not can this assertion be moved into lower_branch and the return value turned back into ()?

Reply:
Well, I thought it more important to keep lower_branch just a pass-through to generated_code::constructor_lower_branch now (like the main lower routine).  Ideally, the lower_branch ISLE term would have a Unit result instead of InstOutput, but that would require a few further changes (like the side_effect constructor no longer works etc.), so I didn't want to do it as part of this patch.   I guess this could be done as a follow-up?

Reply:
Yeah, I think that's a reasonable cleanup to do in another PR.